src/Transport/MessageRetrieved.cs

```diff
@@ -102,14 +105,15 @@ public async Task MoveToErrorQueue(string messageId, string messagePopReceipt, B
 
         void AssertVisibilityTimeout()
         {
-            if (rawMessage.NextVisibleOn != null)
+            if (rawMessage.NextVisibleOn == null)
             {
-                var visibleIn = rawMessage.NextVisibleOn.Value - DateTimeOffset.UtcNow;
-                if (visibleIn < TimeSpan.Zero)
-                {
-                    var visibilityTimeoutExceededBy = -visibleIn;
-                    throw new LeaseTimeoutException(rawMessage, visibilityTimeoutExceededBy);
-                }
+                return;
+            }
+
+            var visibleIn = rawMessage.NextVisibleOn.Value - (messageReceivedTime + timeProvider.GetElapsedTime(startTimestamp));
```

Comment:
Nitpicking: could you extract variables here to make the calculation logic easier to follow? Something like
```suggestion
            var processingTime = timeProvider.GetElapsedTime(startTimestamp);
            var processedAtAccordingToServer = messageReceivedTime + processingTime;
            var visibleIn = rawMessage.NextVisibleOn.Value - processedAtAccordingToServer;
```